uat/src/main/java/com/aws/greengrass/testing/mqtt5/client/MqttReceivedMessage.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.testing.mqtt5.client;
+
+import lombok.Builder;
+import lombok.Data;
+
+/**
+ * Contains information about received MQTT v5.0 message.
+ */
+@Data
+@Builder
+public class MqttReceivedMessage {
```

Comment:
should these classes be generated by protobuf so that they're the same in every language?

Reply:
I decoupled gRPC and MQTT parts by using intermediate objects. It allows replacing or adding another MQTT part much more simply.

Reply:
I'm not sure I see the benefit here, transforming between data types with so many fields opens up opportunities to do the mapping incorrectly. Right now you need to transform from AWS CRT type to intermediate type to gRPC type and all the transforms the other way around too.

"adding another mqtt part" -- another mqtt part of what? If you use the gRPC type as our main data type, then don't you have the same benefit as creating your own intermediate type? I understand not using the AWS CRT type, but the gRPC type is within our control in this package, so that would seem to be OK to me.

Reply:
Another MQTT can be CRT MQTT 3.11 or another MQTT5 implementation on Java.
Idea is do not poison the MQTT code with gRPC types. 
Currently it works in both directions. Sure we have double conversion here.
Benefits: 
1. Code has visible borders.
2. Changes in gRPC protocol do not touch MQTT code (in future multiple implementations). At the moment gRPC proto is the most frequently changed part due to limited time to deep design.
3. Easy to write unit test based on API only objects (not need to check generated code of gRPC)